src/Analytics/Adapter/ActiveCampaign.php

```diff
@@ -40,27 +41,313 @@ class ActiveCampaign extends Adapter
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Delete an account
+     * 
+     * @param string $accountId
+     * 
+     * @return bool
+     */
+    public function deleteAccount(string $accountId): bool
+    {
+        try {
+            $this->call('DELETE', '/api/3/accounts/'.$accountId);
+            return true;
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Sync an association
+     * 
+     * Creates an association if it doesn't exist and updates it if it does
+     * 
+     * @param string $accountId
+     * @param string $contactId
+     * @param string $role
+     * 
+     * @return bool
+     */
+    public function syncAssociation(string $accountId, string $contactId, string $role = ''): bool
+    {
+        // See if the association already exists
+
+        try {
+            $result = $this->call('GET', '/api/3/accountContacts', [], [
+                'filters[account]' => $accountId,
+                'filters[contact]' => $contactId
+            ]);
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+
+        if (intval(json_decode($result, true)['meta']['total']) > 0) {
+            // Update the association
+            $associationId = intval((json_decode($result, true))['accountContacts'][0]['id']);
+
+            try {
+                $result = $this->call('PUT', '/api/3/accountContacts/'.$associationId, [
+                    'Content-Type' => 'application/json',
+                ], [
+                    'accountContact' => [
+                        'jobTitle' => $role
+                    ]
+                ]);
+                return true;
+            } catch (\Exception $e) {
+                $this->logError($e);
+                return false;
+            }
+        } else {
+            // Create the association
+            $result = $this->call('POST', '/api/3/accountContacts', [
+                'Content-Type' => 'application/json',
+            ], ['accountContact' => [
+                'account' => $accountId,
+                'contact' => $contactId,
+                'jobTitle' => $role
+            ]]);
+
+            return true;
+        }
     }
 
     /**
      * @param string $key 
      * @param string $actid
-     * Adapter configuration
+     * @param string $apiKey
+     * @param string $organisationID
      * 
      * @return ActiveCampaign
      */
-    public function __construct(string $key, string $actid)
+    public function __construct(string $key, string $actid, string $apiKey, string $organisationID)
     {
         $this->key = $key;
         $this->actid = $actid;
+        $this->apiKey = $apiKey;
+        $this->endpoint = 'https://'.$organisationID.'.api-us1.com/'; // ActiveCampaign API URL, Refer to https://developers.activecampaign.com/reference/url for more details.
```

Comment:
```suggestion
        $this->endpoint = 'https://' . $organisationID . '.api-us1.com/'; // ActiveCampaign API URL, Refer to https://developers.activecampaign.com/reference/url for more details.
```
When using string concatenation, its better to have a space before and after the period